### PR TITLE
feat(metrics): support backing image metrics

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: https://github.com/longhorn/longhorn/issues/7563

#### What this PR does / why we need it:
Support collect `BackingImage`, `BackupBackingImage` metrics

### CODE REVIEW

**Review by c3y1huang — CHANGES_REQUESTED** [review]

File: metrics_collector/backing_image_collector.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+		}
+	}
+}
+
+func (bc *BackingImageCollector) getDiskNodeMap() (map[string]string, error) {
+	diskNodeMap := make(map[string]string)
+	nodeList, err := bc.ds.ListNodesRO()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get node list")
```
[the remark this answers is not in the document]
Comment (ChanYiLin) [reply metrics_collector/backing_image_collector.go]: Fixed, thanks!

File: metrics_collector/backing_image_collector.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+		bc.logger.WithError(err).Warn("Error during scrape")
+		return
+	}
+
+	for _, backingImage := range backingImageLists {
+		if backingImage.Status.OwnerID == bc.currentNodeID {
+			for diskID, status := range backingImage.Status.DiskFileStatusMap {
+				nodeID, ok := diskNodeMap[diskID]
+				if !ok {
+					bc.logger.Warn("Failed to find nodeID with the given diskID %v of the BackingImage %v", diskID, backingImage.Name)
```
[the remark this answers is not in the document]
Comment (ChanYiLin) [reply metrics_collector/backing_image_collector.go]: Fixed, thanks!

**Review by innobead — DISMISSED** [review]
LGTM. 

Leave to Chinya final review before merging.

File: metrics_collector/metrics_collector.go [hunk 3]
```diff
@@ -44,6 +46,14 @@ func InitMetricsCollectorSystem(logger logrus.FieldLogger, currentNodeID string,
 		logger.WithField("collector", subsystemSnapshot).WithError(err).Warn("Failed to register collector")
 	}
 
+	if err := registry.Register(backingImageCollector); err != nil {
+		logger.WithField("collector", subsystemBackingImage).WithError(err).Warn("Failed to register collector")
+	}
+
+	if err := registry.Register(backupBackingImageCollector); err != nil {
+		logger.WithField("collector", subsystemBackingImage).WithError(err).Warn("Failed to register collector")
```
Comment (c3y1huang) [inline metrics_collector/metrics_collector.go:54]:
```suggestion
		logger.WithField("collector", subsystemBackupBackingImage).WithError(err).Warn("Failed to register collector")
```

**Review by c3y1huang — CHANGES_REQUESTED** [review]

File: metrics_collector/metrics_collector.go (hunk 3 above)
Comment (ChanYiLin) [reply metrics_collector/metrics_collector.go]: Fixed, thanks

**Review by c3y1huang — APPROVED** [review]

**Pull request merged**